getredash/redash
# New ElasticSearch Query Runner

## What type of PR is this? 



## Description


This PR incorporates the excellent work from #4391 with adjustments in #5701. I rebased from `master` and applied both patches to retain everyone's commit attributions.

These changes have been reviewed twice and approved but were never merged. Here's hoping the third time's the charm.

In addition to those commits, I have deprecated the original `elasticsearch` query runner. Existing connections will continue to exist.

## How is this tested?


Unending thanks to @nicolaslm and @wwl717195673 for their unit and manual testing.

I also manually tested the deprecation:
1. Create an `elasticsearch` data source 
2. Apply `deprecated=True` commit and reload UI3. 
3. DS from step 1 is still visible
4. Create `elasticsearch2` database 
5. Check contents of `data_sources` table


<img width="338" alt="CleanShot 2022-07-19 at 21 49 51@2x" src="https://user-images.githubusercontent.com/17067911/179885509-0d3cfbdd-aca5-41fd-a88e-1e78821a2480.png">
<img width="717" alt="CleanShot 2022-07-19 at 21 50 05@2x" src="https://user-images.githubusercontent.com/17067911/179885538-1e8f32e0-34db-4333-9d87-24598ee43e27.png">






## Related Tickets & Documents


## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

#### Data source selection screen:

<img width="524" alt="CleanShot 2022-07-19 at 21 50 24@2x" src="https://user-images.githubusercontent.com/17067911/179885577-34922fce-c567-44bd-802e-6cc3f01c0977.png">
